notebooks/stress-tests.ipynb

```diff
@@ -330,7 +330,7 @@
     "            resp = requests.request(method, url, timeout=abort_timeout, **req_kwargs)\n",
     "            resp.raise_for_status()\n",
     "        except requests.exceptions.Timeout as errT:\n",
-    "            result.codes.append(code)  # read timeout\n",
+    "            result.codes.append(408)  # read timeout\n",
```

Comment:
What was the wrong code for timeout, why not expose it instead of hardcoding 408?  Was hardcoding the previous behavior that you are trying to maintain?

In general, tests are meant to expose errors so masking the error is usually the wrong thing to do.  My 2 cents because I do not really know for sure what is the design decision of Magpie.

Reply:
I'll answer both here, 

I might be wrong on this, please correct me if it is the case. 
But because your request TimeOut after a specified time as `timeout=abort_timeout` it means you dont have any answer from the server, which result in not knowing the response, only knowing the request made, so we can't have the exact Status_Code of the server, so we force a 408 code after the TimeOut limit. And because there is no answer from the Server I force the "Request Timeout" message and I print the `requests.exceptions.Timeout`(request) as `exception_text` message. 

Also, the thing with this test, is that a 408 error happens randomly, so trying to debug this is like flipping a coin, sometime it takes 1 minute sometimes an entire day. So I can try to optimise it on my side, but it would be important that @fmigneault can be unblocked.

Reply:
OHH I misunderstood, so it's the test that is timing out, it's not the actual response from the server.  So in that case, the error reporting should not make it looks like it's a 408 received from the server !

Maybe just put "xxx" as the code and the reason would be "test timeout after ### second"?

So we can differentiate a real timeout as returned by Magpie because the server behind Magpie takes too long to respond vs the test that is timing out because we enforce an arbitrary `abort_timeout`.

But I'd suggest you wait to hear from @fmigneault before changing code.  He knows best how to test Magpie.

Reply:
@ahandan's interpretation is correct. 
We manually set a limit we want to wait for, and if no response from the server is received until then, the client (us) aborts. 
So there is no real code from the server. 
Code 408 is picked because that's the code for "Client Timeout", in contrast to "Server Timeout" that should be 504 (or some other 5xx).

Reply:
Just to be clear, there is no confusion to be had here. 
When the request is initiated, our client establishes a connection with a limit "wait for" x seconds. 
The server accepts that connection and keeps it open for that amount of time to receive the request. 
If either the client fails to produce the request or the server to receive it / read data in time, 408 happens.
In other words, the request did not reach (completely or at all) the server. 
If the server failed to produce a response in time (eg: it itself did not receive a result in time from the service it was supposed to redirect to), 504 happens instead. This is based on the timeout value configured in nginx.

Reply:
OK, it's clear for me now.  Just so it's clear for others as well, can we simply add "(client side)" to the reason "Request TimeOut (client side)"?  Thanks.